.github/workflows/create-builds.yml

```diff
@@ -0,0 +1,112 @@
+# Summary:
+#
+# Reusable workflow for creating the distributable application assets.
+# Will also publish the application if the `app_type` input is `production`.
+#
+name: Create builds
+
+on:
+  workflow_call:
+    inputs:
+      app_type:
+        description: 'Type of application. Valid values are: internal, release-candidate, or production.'
+        required: true
+        type: string
+      build_sha:
+        description: 'SHA of commit to build'
+        required: true
+        type: string
+  workflow_dispatch:
+    inputs:
+      app_type:
+        description: 'Type of application. Valid values are: internal, or release-candidate'
+        required: true
+        type: choice
+        # We only allow internal and release candidate builds when doing manual dispatches.
+        # Production builds should only be triggered by the appropriate calling workflows.
+        options: ['internal', 'release-candidate']
+
+jobs:
+  builds:
+    name: Create builds
+    strategy:
+      matrix:
+        os:
+          [
+            { name: 'linux', image: 'ubuntu-latest' },
+            { name: 'macos', image: 'macos-latest' },
+            { name: 'macos-intel', image: 'macos-13' },
+            { name: 'windows', image: 'windows-latest' },
+          ]
+    runs-on: ${{ matrix.os.image }}
+    outputs:
+      artifact_id: ${{ steps.artifact.outputs.artifact-id }}
+      artifact_url: ${{ steps.artifact.outputs.artifact-url }}
```

Comment:
these are not used at the moment, as they are not really reflective of what we want to provide as an output. Since this workflow runs in a matrix, these outputs will only reflect the value coming from the last-run iteration of the matrix.

What we really want is either:

1. A list of all of the artifact outputs
2. A URL that only points to the general artifacts page i.e. `https://github.com/example-org/example-repo/actions/runs/1/artifacts/`

Reply:
This is no longer relevant, but still am curious about potential ways of achieving something like this, for future reference!